pydeseq2/dds.py

```diff
@@ -317,33 +359,80 @@ def vst(
             )
             # Fit the trend curve with an intercept design
             self.fit_genewise_dispersions()
-            if fit_type == "parametric":
+            if self.fit_type == "parametric":
                 self.fit_dispersion_trend()
 
             # Restore the design matrix and free buffer
             self.obsm["design_matrix"] = self.obsm["design_matrix_buffer"].copy()
             del self.obsm["design_matrix_buffer"]
 
-        # Apply VST
-        if fit_type == "parametric":
+    def vst_transform(self, counts: Optional[np.ndarray] = None) -> np.ndarray:
+        """Apply the variance stabilizing transformation.
+
+        Uses the results from vst_fit method.
+
+        Parameters
+        ----------
+        counts : np.ndarray
+            Counts to transform. If ``None``, use the counts from the current dataset.
+            (default: ``None``).
+
+        Returns
+        -------
+        np.ndarray
+            Variance stabilized counts.
+        """
+
+        if "size_factors" not in self.obsm:
+            raise RuntimeError(
+                "The method vst_fit should be called prior to vst_transform."
+                )
+
+        if counts is None:
+            # the transformed counts will be the current ones
+            normed_counts = self.layers["normed_counts"]
+        else:
+            if self.logmeans is None:
+                # the size factors were still computed iteratively
+                warnings.warn(
+                    "The size factors were fitted iteratively. The size factors will "
+                    "be re-computed with the counts to be transformed. In a train/test "
+                    "fashion with a downstream task, this would result in a leak of "
+                    "data.",
+                    UserWarning
```

Comment:
Add stacklevel=2 this is what pre-commit is complaining about

Reply:
Indeed thanks !